.github/workflows/windows.yml

```diff
@@ -413,66 +413,137 @@ jobs:
+          key: ${{ matrix.plat }}-${{ matrix.name }}-${{ hashFiles('vcpkg/installed/vcpkg/status') }}
+
       - name: 'cmake configure'
         timeout-minutes: 5
         shell: bash
         run: |
-          archgen=${{ matrix.arch }}; [ "${archgen}" = 'x86' ] && archgen='Win32'
-          if [ '${{ matrix.plat }}' = 'uwp' ]; then
-            system='WindowsStore'
-            options='-DCMAKE_SYSTEM_VERSION=10.0'
-          else
-            system='Windows'
-          fi
-          [ '${{ matrix.type }}' = 'Debug' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG='
-          [ '${{ matrix.type }}' = 'Release' ] && options+=' -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE='
-          cmake -B bld ${options} \
-            "-DCMAKE_SYSTEM_NAME=${system}" \
-            -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake \
-            "-DCMAKE_GENERATOR_PLATFORM=${archgen}" \
-            '-DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }}' \
-            -DCMAKE_VS_GLOBALS=TrackFileAccess=false \
-            '-DCMAKE_UNITY_BUILD=${{ matrix.unity }}' \
-            "-DCMAKE_C_FLAGS=${cflags}" \
-            '-DCMAKE_BUILD_TYPE=${{ matrix.type }}' \
+          export VCPKG_ROOT=$PWD/vcpkg
+          cmake . -B bld \
+            -DCMAKE_TOOLCHAIN_FILE=$VCPKG_ROOT/scripts/buildsystems/vcpkg.cmake \
+            -DVCPKG_INSTALLED_DIR=$VCPKG_ROOT/installed \
+            -DVCPKG_TARGET_TRIPLET=${{ matrix.arch }}-${{ matrix.plat }} \
+            -DCMAKE_BUILD_TYPE=${{ matrix.type }} \
             -DCMAKE_UNITY_BUILD=ON \
             -DCURL_WERROR=ON \
             -DBUILD_EXAMPLES=ON \
             -DENABLE_WEBSOCKETS=ON \
-            ${{ matrix.config }} || { cat bld/CMakeFiles/CMake*.yaml; false; }
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG= \
+            -DCMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE= \
+            ${{ matrix.config }}
+
+      - name: Prepare logs on failure
+        if: failure()
+        shell: bash
+        run: |
+          export VCPKG_ROOT=$PWD/vcpkg
+          mkdir -p bld
+          7z a -t7z -r -mx=9 logs.7z \
+              $VCPKG_ROOT/buildtrees/*.log \
+              bld/.ninja_log \
+              bld/build.ninja \
+              bld/install_manifest.txt \
+              bld/vcpkg-manifest-install.log \
+              bld/CMakeFiles/CMake*.yaml
+
+      - name: Upload logs on failure
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: windows_logs_${{matrix.name}}_${{github.event.pull_request.head.sha}}
+          path: logs.7z
 
       - name: 'cmake build'
         timeout-minutes: 5
         shell: bash
         run: |
-          cmake --build bld --config '${{ matrix.type }}' --parallel 5
-          [[ '${{ matrix.config }}' != *'BUILD_SHARED_LIBS=OFF'* ]] && cp -f -p bld/lib/*.dll bld/src/
+          export VCPKG_ROOT="$PWD/vcpkg"
+          cmake --build bld --config ${{ matrix.type }} --parallel 5
```

Comment:
Please keep existing quotes.

Reply:
I remove them because it doing a problem if you try to switch to cmd shell. Sometimes I do that to test thing.
I can convert all the martrix.xxx to environment variables.

Reply:
Keeping changes minimal and serving the CI config (and its shell) should get priority IMO.

Reply:
ok.